code/frameworks/nextjs/src/routing/navigation/index.ts

```diff
@@ -0,0 +1,49 @@
+import type { Mock } from '@storybook/test';
+import { fn } from '@storybook/test';
+import { NextjsRouterMocksNotAvailable } from '@storybook/core-events/preview-errors';
+
+let navigationAPI: {
+  push: Mock;
+  replace: Mock;
+  forward: Mock;
+  back: Mock;
+  prefetch: Mock;
+  refresh: Mock;
+};
+
+export const createNavigation = (overrides: any) => {
+  if (!navigationAPI) {
+    const navigationActions = {
+      push: fn().mockName('nextNavigation.push'),
+      replace: fn().mockName('nextNavigation.replace'),
+      forward: fn().mockName('nextNavigation.forward'),
+      back: fn().mockName('nextNavigation.back'),
+      prefetch: fn().mockName('nextNavigation.prefetch'),
+      refresh: fn().mockName('nextNavigation.refresh'),
+    };
+
+    if (overrides) {
+      Object.keys(navigationActions).forEach((key) => {
+        if (key in overrides) {
+          (navigationActions as any)[key] = fn((...args: any[]) => {
+            return (overrides as any)[key](...args);
+          }).mockName(`nextNavigation.${key}`);
+        }
+      });
+    }
+
+    navigationAPI = navigationActions;
+  }
+
+  return navigationAPI;
+};
+
+export const useRouter = () => {
```

Comment:
I think we should call this `getRouter` as we will re-export `useRouter` once we fully mock next/navigation (including `redirect` etc.)